cclib/io/qcjsonwriter.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2018, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""A writer for MolSSI quantum chemical JSON (QCJSON) files.
+"""
+
+import json
+
+from cclib.parser.utils import convertor
+
+from .cjsonwriter import CJSON as CJSONWriter, JSONIndentEncoder, NumpyAwareJSONEncoder
+
+
+class QCJSONWriter(CJSONWriter):
+    """A writer for QCSchema files."""
+
+    def __init__(self, ccdata, *args, **kwargs):
+        super(QCJSONWriter, self).__init__(ccdata, *args, **kwargs)
+
+    def as_dict(self):
+        qcschema_dict = dict()
+
+        qcschema_dict["schema_name"] = "qcschema_output"
+        qcschema_dict["schema_version"] = 1
+
+        qcschema_dict["molecule"] = dict()
+        qcschema_dict["molecule"]["schema_name"] = "qcschema_molecule"
+        qcschema_dict["molecule"]["schema_version"] = 2
+
+        # FIXME hack
+        qcschema_dict["driver"] = "energy"
```

Comment:
Is the thought here to populate based on `optdone` and the presence of certain attributes or by parsing `metadata->input_file_contents` or something else?

Reply:
I think we need to parse based on what's happened in the output file, rather than what's specified in the input file if possible, since programs can have a complex "path" through them when you specify even simple stuff. For example, if you say "I want a CCSD relaxed dipole moment", which is an "energy" calculation, you'll get the gradient, since calculating the coupled cluster density is expensive compared to calculating the integral derivatives, so your energy calculation has automatically turned into a gradient calculation even though it wasn't explicitly requested.